test/functional/collection_tests.js

```diff
@@ -12,63 +11,64 @@ describe('Collection', function() {
   });
 
-  /**
-   * @ignore
-   */
-  it('should correctly execute basic collection methods', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
+  describe('', function() {
+    let client;
+    let db;
+    beforeEach(function() {
+      client = this.configuration.newClient(this.configuration.writeConcernMax(), {
         poolSize: 1
       });
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+      return client.connect().then(client => {
+        db = client.db(this.configuration.db);
+      });
+    });
+    afterEach(function() {
+      return client.close();
+    });
 
-        db.createCollection('test_collection_methods', function(err, collection) {
+    /**
+     * @ignore
+     */
+    it('should correctly execute basic collection methods', {
+      metadata: {
+        requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
+      },
+
+      // The actual test we wish to run
+      test: function(done) {
+        db.createCollection('test_collection_methods', (err, collection) => {
           // Verify that all the result are correct coming back (should contain the value ok)
-          test.equal('test_collection_methods', collection.collectionName);
+          expect('test_collection_methods').to.equal(collection.collectionName);
           // Let's check that the collection was created correctly
-          db.listCollections().toArray(function(err, documents) {
-            var found = false;
-            documents.forEach(function(document) {
-              if (document.name === 'integration_tests_.test_collection_methods') found = true;
-            });
-            test.ok(true, found);
-
+          db.listCollections().toArray(err => {
+            expect(err).to.not.exist;
             // Rename the collection and check that it's gone
-            db.renameCollection('test_collection_methods', 'test_collection_methods2', function(
-              err
-            ) {
-              test.equal(null, err);
+            db.renameCollection('test_collection_methods', 'test_collection_methods2', err => {
+              expect(err).to.not.exist;
               // Drop the collection and check that it's gone
-              db.dropCollection('test_collection_methods2', function(err, result) {
-                test.equal(true, result);
+              db.dropCollection('test_collection_methods2', (err, result) => {
+                expect(true).to.equal(result);
               });
             });
 
-            db.createCollection('test_collection_methods3', function(err, collection) {
+            db.createCollection('test_collection_methods3', (err, collection) => {
```

Comment:
I think for these asserts we generally want them to be in the format `expect(actual).to.equal(expected)`